kvbc/include/categorization/blocks.h

```diff
@@ -0,0 +1,159 @@
+
+//////////////////////////////////// RAW BLOCKS//////////////////////////////////////
+
+// Reconstructs the updates data as recieved from the user
+// This set methods are overloaded in order to construct the appropriate updates
+
+// Merkle updates reconstruction
+RawBlockMerkleUpdates getRawUpdates(const std::string& category_id,
+                                    const MerkleUpdatesInfo& update_info,
+                                    const BlockId& block_id,
+                                    std::shared_ptr<storage::rocksdb::NativeClient>& native_client) {
+  // For old serialization
+  using namespace concord::kvbc::v2MerkleTree;
+  RawBlockMerkleUpdates data;
+  // Copy state hash (std::copy?)
+  data.state_hash = update_info.state_hash;
+  // Iterate over the keys:
+  // if deleted, add to the deleted set.
+  // else generate a db key, serialize it and
+  // get the value from the corresponding column family
+  for (auto& [key, flag] : update_info.keys) {
+    if (flag.deleted) {
+      data.updates.deletes.push_back(key);
+      continue;
+    }
+    // E.L see how we can optimize the sliver temporary allocation
+    auto db_key =
+        v2MerkleTree::detail::DBKeyManipulator::genDataDbKey(Key(std::string(key)), update_info.state_root_version);
+    auto val = native_client->get(category_id, db_key);
+    if (!val.has_value()) {
+      throw std::logic_error("Couldn't find value for key");
+    }
+    // E.L serializtion of the Merkle to CMF will be in later phase
+    auto dbLeafVal =
+        v2MerkleTree::detail::deserialize<detail::DatabaseLeafValue>(concordUtils::Sliver{std::move(val.value())});
+    ConcordAssert(dbLeafVal.addedInBlockId == block_id);
+
+    data.updates.kv[key] = dbLeafVal.leafNode.value.toString();
+  }
+
+  return data;
+}
+
+// KeyValueUpdatesData updates reconstruction
+RawBlockKeyValueUpdates getRawUpdates(const std::string& category_id,
+                                      const KeyValueUpdatesInfo& update_info,
+                                      const BlockId& block_id,
+                                      std::shared_ptr<storage::rocksdb::NativeClient>& native_client) {
+  RawBlockKeyValueUpdates data;
+  return data;
+}
+
+// shared updates reconstruction
+RawBlockSharedUpdates getRawSharedUpdates(const SharedKeyValueUpdatesInfo& update_info,
+                                          const BlockId& block_id,
+                                          std::shared_ptr<storage::rocksdb::NativeClient>& native_client) {
+  RawBlockSharedUpdates data;
+  return data;
+}
+
+// Raw block is the unit which we can transfer and reconstruct a block in a remote replica.
+// It contains all the relevant updates as recieved from the client in the initial call to storage.
+// It also contains:
+// - parent digest
+// - state hash per category (if exists) E.L check why do we pass it.
+struct RawBlock {
+  RawBlock(const Block& block, std::shared_ptr<storage::rocksdb::NativeClient> native_client) {
```

Comment:
Nit: Maybe we don't need to bump the ref count on the native client (i.e. use non-owning pointer, as mentioned above).

Reply:
yep, I used to work like this, but some people didn't like it :)

Reply:
Bad people :) I mean we just use the client in the scope, we don't save it as a member or anything. I think that's the typical pattern for C++. If we were saving it somewhere that's a different topic, but this ctor is just a function.

Better yet - we can use a reference! :)